testing: assert ErrorContains/ErrorIs instead of Error

cce opened the pull request:
WIP

cce committed:
errorcontains and replace tools

cce committed:
test: replace Error assertions with errorcontains.CaptureError

Phase 1 of error assertion migration: Replace all require.Error,
assert.Error, and a.Error calls with errorcontains.CaptureError
to capture actual error types and messages during test runs.

This enables analysis of what errors are actually being checked,
so they can be converted to more specific assertions like
ErrorContains, ErrorIs, or ErrorAs in Phase 2.

108 files changed, ~550 replacements.

cce committed:
test: use PID-specific output files to avoid contention

cce committed:
errorcontains phase 3 tooling

cce committed:
test: update aggregate tool to use go-lcss library

Use the go-lcss library for efficient multi-string longest common
substring computation instead of the flawed iterative approach.

cce committed:
test: convert CaptureError to ErrorContains/ErrorIs

Convert 345 errorcontains.CaptureError calls to specific assertions:
- 36 removed (kept existing ErrorIs)
- 8 converted to ErrorIs (from Equal with sentinel)
- 30 converted to ErrorContains (from Contains)
- 273 standalone converted to ErrorContains with LCS substrings

210 calls remain for manual review.

cce committed:
test: convert more CaptureError calls to ErrorContains

Continue converting CaptureError/CaptureErrorAssert calls to specific
assertions based on captured error data. This batch includes:

- agreement: bundle, certificate, vote tests
- crypto: batchverifier, multisig tests
- data: participationRegistry, bookkeeping, pools tests
- data/transactions: verify, txnBatch, verifiedTxnCache tests
- ledger: apply/payment, boxtxn, ledger tests
- network: p2pMetainfo tests
- node: follower_node tests
- rpcs: txSyncer tests

Also fixes path normalization in aggregate/phase3 tools to handle
both go-algorand and errorcontains repo names.

cce committed:
test: convert remaining CaptureError calls to ErrorContains

Convert all non-E2E CaptureError/CaptureErrorAssert calls to specific
error assertions across 18 test files:

- agreement: proposalTracker_test.go, pseudonode_test.go
- cmd/goal: node_test.go
- data: ledger_test.go
- ledger: acctupdates_test.go, ledger_test.go
- network: addr_test.go, wsNetwork_test.go
- stateproof: stateproof_test.go
- tools: block-generator, dnssec, resolveController, typeAnalyzer
- util: rateLimit_test.go

Each assertion now checks for specific error message substrings
rather than just verifying an error occurred. Two cases use
require.Error with comments where error messages vary by
platform or external factors.

Also removes unused errorcontains imports.

cce committed:
test: fix struct dump artifacts in ErrorContains assertions

Replace meaningless struct dump fragments (from LCS algorithm) with
actual error message substrings in ErrorContains assertions:

- ledger_test.go: Fix assertions for invalid tx type, overspend,
  no signature, applyData mismatch, and overlapping lease
- bundle_test.go: Fix duplicated equivocation vote assertion
- Remove unused errorcontains imports after converting all calls

cce committed:
fix: correct error assertion bugs found during review

- crypto/multisig_test.go: fix parameter order in ErrorContains
- config/config_test.go: remove machine-specific temp paths
- netdeploy/networkTemplates_test.go: remove absolute path prefix
- network/p2p/p2p_test.go: use generic error prefix for all test cases
- ledger/double_test.go: remove overly-specific logic eval error check
- tools/network/resolver_test.go: remove machine-specific IPv6 resolver

cce committed:
fix: replace struct dump artifact with overspend assertion

The `, data` substring was matching part of a struct dump in the error
message, not a meaningful error condition. These tests should assert
`overspend` which is the actual error type being returned.

cce committed:
fix: correct more error assertion issues from CI failures

- Remove backslash-escaping in raw strings (quotes need no escaping)
- Revert generic error checks where various errors can occur
- Fix parameter order in ErrorContains calls
- Remove dynamic values from error substrings (round numbers, etc.)

cce committed:
fix: remove dynamic round/step values from error assertions

The round, step, and dbRound values vary between test runs, so only
check for the stable error message prefix.

cce committed:
fix: address lint issues in errorcontains package

cce committed:
switch to deadlock.Mutex

cce committed:
Merge remote-tracking branch 'upstream/master' into assert-errorcontains

jannotti commented on ledger/eval_simple_test.go:
```diff
@@ -87,14 +87,14 @@ func TestBlockEvaluator(t *testing.T) {
 	txgroup := []transactions.SignedTxn{stbad}
 	err = eval.TestTransactionGroup(txgroup)
-	require.Error(t, err)
+	require.ErrorContains(t, err, `transaction already in ledger`)
```
Definitely not the intent.  This is a bad test that is trying to test signature problems with `TestTransactionGroup`, which doesn't check signatures.

(The test above is also misleading, because the signature is not checked)

cce committed:
better / more specific error strings

cce committed:
run modernize

cce committed:
fix contains

cce committed:
run make tidy

cce committed:
add errorcontains output to artifacts

cce committed:
Merge remote-tracking branch 'upstream/master' into assert-errorcontains

cce committed:
Merge remote-tracking branch 'upstream/master' into assert-errorcontains

cce committed:
Merge upstream/master

cce committed:
tests: convert ErrorContains to ErrorIs where sentinel errors exist

Add sentinel tool that scans for error sentinel definitions and converts
require.ErrorContains calls to require.ErrorIs where the error message
matches a known sentinel. Supports both exact and substring matching,
and handles both exported and unexported sentinels.

Converts 78 assertions across 20 test files.

cce committed:
network/addr: add errMultiaddrParse sentinel and use table-driven test

Wrap multiaddr library errors with a sentinel so they can be checked
with ErrorIs. Convert the test to use a table with expected error types.

cce committed:
tests: convert ErrorContains to ErrorAs where typed errors exist

Use require.ErrorAs for 64 call sites across 24 test files where
runtime error captures showed typed errors (e.g. *strconv.NumError,
base64.CorruptInputError, *ledgercore.TransactionInLedgerError)
rather than plain error strings.

cce committed:
ledger: convert remaining ErrorContains to ErrorAs for errTxTailMissingRound

Two CheckDup calls had function arguments with commas that broke the
automated regex matching. Convert manually to match the existing pattern.

cce committed:
tests: use assertion object style consistently

When a test uses a := require.New(t) or a := assert.New(t), use
a.ErrorIs/ErrorContains/ErrorAs instead of require.ErrorIs(t, ...).
This maintains consistent style within each test function.
